### DISCUSSION TITLE
Developer Style Guidelines

### DISCUSSION THREAD

**Discussion by HexSleeves** [blue-build/.github#6] [General]
https://github.com/blue-build/cli/pull/28#pullrequestreview-1849930658

This discussion is to lock down our styling guide to use across all of blue-build. We should have a core repo with all our linting, formatting, etc configs that can be easily distributed to our repos. This is starting the discussing!

**Comment by xynydev** [comment]
I 100% support the effort. All repos don't need all the linting & formatting though, but it's probably alright to put those in one repo _just in case_ more rust or bash or webdev repos are created.

**Comment by gmpinder** [comment]
One thing I noticed while we were working on the cli tool, @bayou-brogrammer you seem to like putting root module files as `mod.rs`. I know that this used to be the de-facto way to setup modules in Rust, but now it's [suggested](https://doc.rust-lang.org/book/ch07-05-separating-modules-into-different-files.html#alternate-file-paths) to move away from doing  it like that. 

I personally would like to go with the new style rather than the old. I'd like your thoughts on this.

**Reply by HexSleeves** [reply]
That's fine. Both have their pros and cons. In some instances it makes the root dir cleaner without having to pollute the root directory. I'm fine either way.

Let's draft up a style guide to put in the repo so we always have a reference point
